### PR TITLE
Make queueing rules pluggable

### PR DESCRIPTION
**Pull request by cberner**
First step toward: https://github.com/prestodb/presto/issues/4742

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java [hunk 1]
```diff
@@ -16,8 +16,11 @@
 import io.airlift.configuration.Config;
 import io.airlift.configuration.LegacyConfig;
 
+import javax.validation.constraints.NotNull;
+
 public class FeaturesConfig
 {
+    public static final String DEFAULT_RESOURCE_GROUP_MANAGER = "default";
```
Comment (dain) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: Maybe we can come up with a better name than `default`.. something that describes the rules system

**Comment by dain** [comment]
Looks good

**Comment by dain** [comment]
looks good

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java (hunk 1 above)
Comment (cberner) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java]: Good point. I'll change it to "file"

**Pull request merged**